test/form/samples/compact/_expected/amd.js

```diff
@@ -0,0 +1,4 @@
+define(['external'],function(x){'use strict';x=x&&x.hasOwnProperty('default')?x['default']:x;var self = {get default(){returnfoo$$1}};if(typeof Symbol!=='undefined'&&Symbol.toStringTag)Object.defineProperty(self,Symbol.toStringTag,{value:'Module'});elseObject.defineProperty(self,'toString',{value:function(){return'[object Module]';}});/*#__PURE__*/Object.freeze(self);console.log(self);
+function foo$$1 () {
+	console.log( x );
+}return foo$$1;});
```

Comment:
Also it would be nice to separate the user code from the wrapper via new-lines. Not only is it more consistent with the concept of only compactifying the auto-generated code, it will also break down if the user's code ends with a line comment (would not hurt to add this to the test as well)

Reply:
Good point. Let me rather add the ability to detect and remove trailing line comments.

The intention here is that a plugin can perform minification in the transform hook to get an overall minification build running over per-module minify workers.

Reply:
But trailing line-comments could also be a purposefully placed footer comment. It might be ok to remove it in some situations but we actually have logic to NOT do this as it is still one goal for rollup to preserve all comments a user added as long as we think they do not describe removed code. If a minifier is doing its thing, it will remove those comments anyway so the presence of such a comment would signal that minification was NOT performed and is NOT the intended goal here.

And line comments are just an example. If you throw a line such as `const x = 3 return foo;` into acorn, it will tell you it is invalid syntax. So you also need to make sure there is a semicolon. Etc. etc.

All in all I am not happy about this approach and would prefer the line-break.

Reply:
On the other hand, there is no reason to keep the semicolon after `return foo$$1`

Reply:
Oh, reading the actual code you wrote, you actually DO preserve the trailing comments. So I am cool with that. Though I am not sure we should do it

Nevertheless, we should also detect for the situation where the last line is non-empty, does not contain a line-comment and where the last non-whitespace character is not a semicolon in which cases I would suggest to append a semicolon instead.